scripts/main/contextMenu.js

```diff
@@ -27,38 +30,53 @@ contextMenu.add = function (e) {
 		items.splice(3, 1);
 	}
 
-	// prepend further buttons if menu bar is reduced on small screens
-	let button_visibility_album = $("#button_visibility_album");
-	if (button_visibility_album && button_visibility_album.css("display") === "none") {
-		items.unshift({
-			title: build.iconic("eye") + lychee.locale["VISIBILITY_ALBUM"],
-			visible: lychee.enable_button_visibility,
-			fn: (event) => album.setPublic(album.getID(), event),
-		});
-	}
-	let button_trash_album = $("#button_trash_album");
-	if (button_trash_album && button_trash_album.css("display") === "none") {
-		items.unshift({
-			title: build.iconic("trash") + lychee.locale["DELETE_ALBUM"],
-			visible: lychee.enable_button_trash,
-			fn: () => album.delete([album.getID()]),
-		});
-	}
-	let button_move_album = $("#button_move_album");
-	if (button_move_album && button_move_album.css("display") === "none") {
-		items.unshift({
-			title: build.iconic("folder") + lychee.locale["MOVE_ALBUM"],
-			visible: lychee.enable_button_move,
-			fn: (event) => contextMenu.move([album.getID()], event, album.setAlbum, "ROOT", album.getParent() !== ""),
-		});
-	}
-	let button_nsfw_album = $("#button_nsfw_album");
-	if (button_nsfw_album && button_nsfw_album.css("display") === "none") {
-		items.unshift({
-			title: build.iconic("warning") + lychee.locale["ALBUM_MARK_NSFW"],
-			visible: true,
-			fn: () => album.setNSFW(album.getID()),
-		});
+	if (visible.album() && album.isUploadable()) {
+		// prepend further buttons if menu bar is reduced on small screens
+		let albumID = album.getID();
+		if (album.isTagAlbum()) {
+			// For tag albums the context menu is normally not used.
+			items = [];
+		}
+		if (albumID !== "starred" && albumID !== "public" && albumID !== "recent") {
```

Comment:
I wonder how stable this condition is with respect to future changes or additions. May gut feelings tells me that the condition would be more reliable, if it dit not exclude certain albums which don't support the actions but if it included albums which do support the actions explicitly.

Proposal: `if( Number.isInteger(albumID) || albumID === "unsorted" )`.

Reply:
I like your suggestion though I needed to add a `parseInt` in there as well since `albumID` is always a string on `master` (I seem to remember that that might change with your refactoring though?).